test/api/fixtures.go

```diff
@@ -0,0 +1,524 @@
+/*
+Copyright 2024-2025 the Unikorn Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//nolint:revive,staticcheck // dot imports are standard for Ginkgo/Gomega test code
+package api
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// ClusterPayloadBuilder builds cluster payloads for testing.
+type ClusterPayloadBuilder struct {
+	payload map[string]interface{}
+	config  *TestConfig
+}
+
+// NewClusterPayload creates a new cluster payload builder with defaults from config.
+func NewClusterPayload() *ClusterPayloadBuilder {
+	config := LoadTestConfig()
+	timestamp := time.Now().Format("20060102-150405")
+	uniqueName := fmt.Sprintf("testautomationcreate-%s", timestamp)
+
+	return &ClusterPayloadBuilder{
```

Comment:
Is there some reason you aren't using the type safe versions 90% of the time then converting to JSON for the request/injecting bad stuff?  Smells like a breeding ground for mistakes IMO.